graphium/data/datamodule.py

```diff
@@ -821,23 +814,16 @@ def __init__(
                 - `df_path`
                 - `smiles_col`
                 - `label_cols`
-            dataloading_from: Whether to load the data from RAM or from disk. If set to "disk", the data
-                must have been previously cached with `processed_graph_data_path` set. If set to "ram", the data
-                will be loaded in RAM and the `processed_graph_data_path` will be ignored.
             featurization: args to apply to the SMILES to Graph featurizer.
             batch_size_training: batch size for training and val dataset.
             batch_size_inference: batch size for test dataset.
             num_workers: Number of workers for the dataloader. Use -1 to use all available
                 cores.
             pin_memory: Whether to pin on paginated CPU memory for the dataloader.
-            featurization_n_jobs: Number of cores to use for the featurization.
```

Comment:
The only thing that's currently parallelized in C++ is the part that computes InChI keys during preprocessing.  It currently uses 1 thread per CPU core (including any hyperthreading) in most cases.  If you think it'd be worth having that be controllable, I could add back an option for it, maybe with a different name, since it's not featurization anymore.  When it's parallelized, it is only for a short time compared to reading the label data from the input files or training, e.g. 2 seconds for ToyMix or maybe about 1 minute for LargeMix on a CPU with 32 logical cores (16 physical cores), so it might be overkill to have a parameter for it.

The parallelization of batch preparation during training is still managed by PyTorch Lightning using separate processes, so I haven't changed that part.

Reply:
I've added the option as `preprocessing_n_jobs`, with `0` indicating to have a thread for all cores, and `-1` indicating to have a thread for all cores except 1, etc, to avoid other processes starving, if that's a concern for the user, such as in an interactive setting.